oximeter/types/src/histogram.rs

```diff
@@ -1191,6 +1191,65 @@ where
     }
 }
 
+pub trait Bits: Integer {
+    const BITS: u32;
+    fn next_power(self) -> Option<Self>;
+}
+
+macro_rules! impl_bits {
+    ($type_:ty) => {
+        impl Bits for $type_ {
+            const BITS: u32 = Self::BITS;
+
+            fn next_power(self) -> Option<Self> {
+                self.checked_mul(2)
+            }
+        }
+    };
+}
+
+impl_bits!(u8);
+impl_bits!(u16);
+impl_bits!(u32);
+impl_bits!(u64);
+
+/// A trait for generating logarithmically-spaced bins.
```

Comment:
it might be worth mentioning here that this is used for the `Histogram::power_of_two` constructor?

Reply:
also, I'm not personally convinced this needs to be a separate trait --- why not just do the bin construction in the `Histogram::power_of_two` constructor, and just require the `Bits` trait? having a whole additional trait for it kinda just feels like noise that makes the public API more complex, to me?

Reply:
Fair point, I can put that just in the `Histogram` impls for integer bin types.

Reply:
Done in bc6f0d010. We can also make this constructor infallible :)